jose/Jwa.ml

```diff
@@ -1,27 +1,70 @@
-type kty = [ `oct | `RSA | `EC ]
+type kty =
+  [ `oct  (** Octet sequence (used to represent symmetric keys) *)
+  | `RSA  (** RSA  *)
+  | `EC  (** Elliptic Curve *) ]
 
-let kty_to_string = function `oct -> "oct" | `RSA -> "RSA" | `EC -> "EC"
+let kty_to_string : kty -> string = function
+  | `oct -> "oct"
+  | `RSA -> "RSA"
+  | `EC -> "EC"
 
-let kty_of_string = function
+let kty_of_string : string -> kty = function
   | "oct" -> `oct
   | "RSA" -> `RSA
   | "EC" -> `EC
-  | _ -> `RSA
+  | str -> raise (Failure ("Unsupported kty: " ^ str))
 
-type alg = [ `RS256 | `HS256 | `none | `Unknown ]
+type alg =
```

Comment:
why change these to variants instead of keeping them polyvariants?

Reply:
I reverted this and changed enc to be a poly variant as well